person.rb

```diff
@@ -12,26 +13,84 @@ def initialize(age, name, id, parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(book, date)
+    Rental.new(date, self, book)
+  end
+
+  def correct_name
+    @name
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
-  def of_age?
-    @age >= 18
+  def self.create_student
+    print 'Classroom: '
+    classroom = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission_user = gets.chomp
+    has_permission = case parent_permission_user.downcase
+                     when 'y'
+                       true
+                     else
+                       false
+                     end
+
+    student = Person.new(classroom, age, name, parent_permission: has_permission)
+    persons << student
+
+    puts 'Student created'
   end
 
-  def correct_name
-    @name
+  def self.create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Person.new(specialization, age, name)
+    persons << teacher
+
+    puts 'Teacher created'
   end
-end
 
-person = Person.new(22, 'maximilianus', 1)
-person.correct_name
-capitalized_person = CapitalizeDecorator.new(person)
-capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-capitalized_trimmed_person.correct_name
+  def self.create_person(persons)
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    option = gets.chomp.to_i
+    create_student.call
+    create_teacher.call
+    case option
```

Comment:
You're doing great so far, but just as the previous reviewer highlighted, the creation of a person is still giving errors and crashes the program. 

- [ ] When I'm creating a person, a student, or a teacher, the error comes from this section of your code. Kindly check this method and fix the error given when selecting either Y/N for parent permission for every creation of a person.

**See the screenshot below**
![image](https://user-images.githubusercontent.com/50941074/198748480-f24e57af-315c-4410-b543-b6ca2c2fabda.png)